src/geom/TileMatrix.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2015-2018 WorldWind Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * @exports TileMatrix
+ */
+define([
+        '../error/ArgumentError',
+        '../util/Logger',
+        '../geom/Sector'
+    ],
+    function (ArgumentError,
+              Logger,
+              Sector) {
+        "use strict";
+
+        /**
+         * TileMatrix is a collection of tiles for a fixed resolution (degrees per pixel). It contains the dimensions of
+         * the tile in pixels and the dimension of the matrix in tiles.
+         * @param sector the geographic coverage of this TileMatrix
+         * @param ordinal the index of this TileMatrix in the TileMatrixSet
+         * @param matrixWidth the number of tiles in the x direction
+         * @param matrixHeight the number of tiles in the y direction
+         * @param tileWidth the number of pixels or points in the x direction
+         * @param tileHeight the number of pixels or points in the y direction
+         * @constructor
+         */
+        var TileMatrix = function (sector, ordinal, matrixWidth, matrixHeight, tileWidth, tileHeight) {
+            if (!sector) {
+                throw new ArgumentError(
+                    Logger.logMessage(Logger.LEVEL_SEVERE, "TileMatrix", "constructor",
+                        "missingSector"));
+            }
+
+            if (ordinal < 0) {
```

Comment:
These numeric tests pass if the variable is undefined which, unfortunately, is a likely error case if you forget to specify all the parameters.
You could add some unit tests to validate this e.g.
        it("test", function () {
            expect(function () {
                TileMatrixSet.fromTilePyramid(sector);
            }).toThrow();
        });

Reply:
I started with a more explicit check (e.g. typeof) but didn't see that pattern in other geom objects so I reverted to the bounds checking.

Reply:
Numeric argument checks are intentionally anything other than range checks. Early on we took the position that it isn't worth the trouble it takes to meaningfully protect the caller from themselves in this JavaScript. It's an adjustment I've seen every seasoned Java developer go through.